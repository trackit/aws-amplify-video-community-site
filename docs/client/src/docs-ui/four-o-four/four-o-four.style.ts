import {css} from "emotion";
import {MQDesktop} from "../../amplify-ui/styles/media";

export const hostStyle = css`
  width: 100%;
  height: calc(100vh - 10.75rem);
  padding: 1rem;

  ${MQDesktop} {
    height: calc(100vh - 7.75rem);
  }

  > div {
    width: 100%;
    height: 100%;
    text-align: center;
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;

    h1 {
      margin-bottom: 0.5rem;
    }

    p {
      margin-bottom: 1rem;
      max-width: 32rem;
    }
  }
`;
