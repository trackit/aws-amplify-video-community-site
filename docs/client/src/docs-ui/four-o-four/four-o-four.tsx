import {Component, h, Host} from "@stencil/core";
import {hostStyle} from "./four-o-four.style";

@Component({tag: "docs-four-o-four", shadow: false})
export class DocsFourOFour {
  render() {
    return (
      <Host class={hostStyle}>
        <div>
          <h1>404</h1>
          <p>
            {`Apologies––we can't seem to find the page for which you're looking. If this is a mistake, please `}
            <amplify-external-link
              href={`https://github.com/aws-amplify/docs/issues/new?title=[missing-page]&labels=v2&body=${encodeURI(
                `**Page**: [\`${location.href}\`](${location.href})

**Feedback**: <!-- your feedback here -->
`,
              )}`}
            >
              file an issue
            </amplify-external-link>
            {` and we'll fix it ASAP.`}
          </p>
          <docs-internal-link-button href="/">
            Return to the landing page
          </docs-internal-link-button>
        </div>
      </Host>
    );
  }
}
