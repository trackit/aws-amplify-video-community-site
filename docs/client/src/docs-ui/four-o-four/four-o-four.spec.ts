import {DocsFourOFour} from "./four-o-four";
import {newSpecPage} from "@stencil/core/testing";

describe("docs-four-o-four", () => {
  describe("Render logic", () => {
    it("should render", async () => {
      expect(
        (
          await newSpecPage({
            components: [DocsFourOFour],
            html: `<docs-four-o-four />`,
          })
        ).root,
      ).toMatchSnapshot();
    });
  });
});
