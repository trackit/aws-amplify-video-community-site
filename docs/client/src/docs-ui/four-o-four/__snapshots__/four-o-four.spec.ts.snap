// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`docs-four-o-four Render logic should render 1`] = `
<docs-four-o-four class="css-it9o5g">
  <div>
    <h1>
      404
    </h1>
    <p>
      Apologies––we can't seem to find the page for which you're looking. If this is a mistake, please
      <amplify-external-link href="https://github.com/aws-amplify/docs/issues/new?title=[missing-page]&amp;labels=v2&amp;body=**Page**:%20%5B%60http://testing.stenciljs.com/%60%5D(http://testing.stenciljs.com/)%0A%0A**Feedback**:%20%3C!--%20your%20feedback%20here%20--%3E%0A">
        file an issue
      </amplify-external-link>
      and we'll fix it ASAP.
    </p>
    <docs-internal-link-button href="/">
      Return to the landing page
    </docs-internal-link-button>
  </div>
</docs-four-o-four>
`;
